### PR TITLE
[DDCE-2719] Migrated to HMRC-Mongo

### PR DESCRIPTION
**Pull request by matthewwardcap**
Separate ticket may need adding to update `play-frontend-hmrc`

- Cleaned up dependencies
- Performed migration
- Swapped mockito
- Added repository test

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -3,29 +3,29 @@ import sbt._
...
 
-  val test = Seq(
-    "org.scalatest"               %% "scalatest"                % "3.2.12",
+  val test: Seq[ModuleID] = Seq(
+    "uk.gov.hmrc.mongo"           %% "hmrc-mongo-test-play-28"  % "0.73.0",
+    "org.scalatest"               %% "scalatest"                % "3.2.14",
     "org.scalatestplus.play"      %% "scalatestplus-play"       % "5.1.0",
     "org.scalatestplus"           %% "scalatestplus-scalacheck" % "3.1.0.0-RC2",
-    "org.scalatestplus"           %% "mockito-3-12"             % "3.2.10.0",
     "org.pegdown"                 %  "pegdown"                  % "1.6.0",
```
Comment (frikit) [inline project/AppDependencies.scala:21]: can be deleted

**Review by frikit — CHANGES_REQUESTED** [review]
- Fix deprecated warnings in build.sbt related to `in`
- Fix depreacted warnings in build.sbt for covearge settings
- Remove repositories from `ScoverageKeys.coverageExcludedFiles` as we should cover them and you added tests
- Delete lines 42-46 in build sbt
- Remove all dependency overrides as most likely you dont need them

File: app/repositories/SessionRepository.scala [hunk 2]
```diff
@@ -16,72 +16,60 @@
...
-      }
-    }
+    collection.replaceOne(selector, newUser, replaceOptions).headOption().map(_.exists(_.wasAcknowledged()))
   }
-}
-
-trait SessionRepository {
 
-  def get(id: String): Future[Option[UserAnswers]]
+//  def set(userAnswers: UserAnswers): Future[Boolean] = {
```
[the remark this answers is not in the document]
Comment (matthewwardcap) [reply app/repositories/SessionRepository.scala:64]: Done

File: app/config/FrontendAppConfig.scala [hunk 3]
```diff
@@ -71,4 +71,7 @@ class FrontendAppConfig @Inject() (configuration: Configuration,
   private val maxYear: Int = configuration.get[Int]("dates.maximum.year")
   lazy val maxDate: LocalDate = LocalDate.of(maxYear, maxMonth, maxDay)
 
+  lazy val cachettlSessionInSeconds: Long = configuration.get[Int]("mongodb.timeToLiveInSeconds")
+  lazy val dropIndexes: Boolean = configuration.getOptional[Boolean]("microservice.services.features.mongo.dropIndexes").getOrElse(true)
```
[the remark this answers is not in the document]
Comment (matthewwardcap) [reply app/config/FrontendAppConfig.scala:75]: Done

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -3,29 +3,29 @@ import sbt._
...
+    "uk.gov.hmrc.mongo"   %% "hmrc-mongo-play-28"             % "0.73.0",
     "uk.gov.hmrc"         %% "play-frontend-hmrc"             % "1.1.0-play-28",
-    "uk.gov.hmrc"         %% "play-conditional-form-mapping"  % "1.9.0-play-28",
-    "uk.gov.hmrc"         %% "bootstrap-frontend-play-28"     % "5.24.0",
-    "com.typesafe.play"   %% "play-json-joda"                 % "2.7.4",
-    "uk.gov.hmrc"         %% "domain"                         % "6.2.0-play-28"
+    "uk.gov.hmrc"         %% "play-conditional-form-mapping"  % "1.12.0-play-28",
+    "uk.gov.hmrc"         %% "bootstrap-frontend-play-28"     % "7.8.0",
+    "com.typesafe.play"   %% "play-json-joda"                 % "2.9.3",
+    "uk.gov.hmrc"         %% "domain"                         % "8.1.0-play-28"
```
[the remark this answers is not in the document]
Comment (matthewwardcap) [reply project/AppDependencies.scala:13]: Seems to be still in use in quite a few files

File: project/AppDependencies.scala (hunk 1 above)
Comment (matthewwardcap) [reply project/AppDependencies.scala:21]: Done

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -3,29 +3,29 @@ import sbt._
...
-    "org.reactivemongo"   %% "play2-reactivemongo"            % "0.20.13-play28",
+    "uk.gov.hmrc.mongo"   %% "hmrc-mongo-play-28"             % "0.73.0",
     "uk.gov.hmrc"         %% "play-frontend-hmrc"             % "1.1.0-play-28",
-    "uk.gov.hmrc"         %% "play-conditional-form-mapping"  % "1.9.0-play-28",
-    "uk.gov.hmrc"         %% "bootstrap-frontend-play-28"     % "5.24.0",
-    "com.typesafe.play"   %% "play-json-joda"                 % "2.7.4",
-    "uk.gov.hmrc"         %% "domain"                         % "6.2.0-play-28"
+    "uk.gov.hmrc"         %% "play-conditional-form-mapping"  % "1.12.0-play-28",
+    "uk.gov.hmrc"         %% "bootstrap-frontend-play-28"     % "7.8.0",
+    "com.typesafe.play"   %% "play-json-joda"                 % "2.9.3",
```
[the remark this answers is not in the document]
Comment (matthewwardcap) [reply project/AppDependencies.scala:12]: Done

**Comment by frikit** [comment]
Please update hmrc frontend anyway will do testing and changes are very simple. I will help and list them here.
- Update deps to latest
- fix selector for page name
- delete govuk route

You can see changes in this PR https://github.com/hmrc/register-estate-deceased-settlor-frontend/pull/82/files

**Review by frikit — APPROVED** [review]

**Pull request merged**